src/irmin-pack/pack_index.ml

```diff
@@ -10,27 +10,54 @@
    ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE. *)
 
-module type S = Index.S with type value = int64 * int * char
+module type S = sig
+  type t
+
+  type key
+
+  type value = int64 * int * char
+
+  val v :
+    ?fresh:bool ->
+    ?readonly:bool ->
+    ?shared:bool ->
+    log_size:int ->
+    fan_out_size:int ->
+    string ->
+    t
+
+  val clear : t -> unit
+
+  val flush : t -> unit
+
+  val add : t -> key -> value -> unit
+
+  val find_all : t -> key -> value list
+end
 
 module Make (K : Irmin.Hash.S) = struct
   module Key = struct
-    type t = K.t
+    type t = string
 
-    let pp = Irmin.Type.pp K.t
+    let pp = Fmt.fmt "%S"
 
-    let hash = K.short_hash
+    let hash = Hashtbl.hash
 
-    let hash_size = 63
+    let hash_size = Sys.int_size - 1
```

Comment:
Sorry for the late review. 
I think `Hashtbl.hash` only gives 30 bits ints, so `hash_size` should be `30` here.